### PR TITLE
Class to function component change for Visit page

### PR DESCRIPTION
**Pull request by tejaswip-msft**
# What
Moving from Class to function component for Visit page

# Why
existing PR: https://github.com/Azure-Samples/communication-services-virtual-visits-sample/pull/102
Breaking down the PR for easier review

# How Tested
Unit Tests
Manual test once JoinTeamsMeeting page is also changed to function components

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: client/src/Visit.test.tsx [hunk 1]
```diff
@@ -74,8 +56,9 @@ describe('Visit', () => {
     expect(headers.length).toBe(0);
   });
 
-  it('renders a generic error UI when config throws an error', async () => {
-    (fetchConfig as jest.Mock).mockImplementation(
+  it('renders an generic error UI when config throws an error', async () => {
```
Comment (ralphgabrinao) [inline client/src/Visit.test.tsx:59]: Can we also add a test case for when `fetchToken` throws an error?

File: client/src/components/MeetingExperience.tsx [hunk 2]
```diff
@@ -140,19 +137,15 @@ const _createCustomAdapter = async (
...
-  const threadId = getChatThreadIdFromTeamsLink(locator.meetingLink);
-
-  const chatClient = chatEnabled
-    ? createStatefulChatClient({
-        userId,
-        displayName,
-        endpoint,
-        credential
-      })
-    : createStubChatClient(userId, threadId);
```
Comment (ralphgabrinao) [inline client/src/components/MeetingExperience.tsx:152]: This `chatEnabled` conditional is a recent change that we should keep: https://github.com/Azure-Samples/communication-services-virtual-visits-js/pull/43/files#diff-98e77a3eb2de1327d4cdf34141837b5ca45fe5dad35a2447a6e0ec3eafb272b6

File: client/src/components/MeetingExperience.tsx [hunk 3]
```diff
@@ -73,58 +71,57 @@ export const MeetingExperience = (props: MeetingExperienceProps): JSX.Element =>
       }
     };
 
-    _createAdapters();
-  }, [credential, displayName, endpointUrl, locator, userId, onDisplayError]);
-
-  if (callWithChatAdapter) {
```
Comment (ralphgabrinao) [inline client/src/components/MeetingExperience.tsx:79]: I think we can leave this component as is and revert the changes. This `MeetingExperience` component was already converted to functional components in this PR: https://github.com/Azure-Samples/communication-services-virtual-visits-js/pull/28

**Review by ralphgabrinao — DISMISSED** [review]

File: client/src/components/MeetingExperience.tsx (hunk 3 above)
Comment (tejaswip-msft) [reply client/src/components/MeetingExperience.tsx:79]: I can see that the PR is merged, but i am not able to see the changes in main 
https://github.com/Azure-Samples/communication-services-virtual-visits-sample/blob/main/client/src/components/MeetingExperience.tsx

Comment (ralphgabrinao) [reply client/src/components/MeetingExperience.tsx:79]: ^ -- It is because the PR was done in the new repo (this repo) -- https://github.com/Azure-Samples/communication-services-virtual-visits-js/blob/main/client/src/components/MeetingExperience.tsx

So MeetingExperience is now a function component

**Comment by yilunqian-msft** [comment]
Considered we are using functional components, hooks and eslint, probably it's a good time to install this dependency: https://www.npmjs.com/package/eslint-plugin-react-hooks#installation.

**Review by hansung-msft — APPROVED** [review]
It looks good to me. Done e2e it worked fine.
